ocp_utilities/infra.py

```diff
@@ -49,6 +53,35 @@ def assert_nodes_schedulable(nodes):
         )
 
 
+def assert_pods_failed_or_pending(pods: list, namespace=None) -> None:
+    """
+    Validates all pods are not in failed nor pending phase
+
+    Args:
+         pods: List of pod objects
+
+    Raises:
+        PodsFailedOrPendingError, if there are failed or pending pods
+    """
+    LOGGER.info("Verify all pods are not failed nor pending.")
+
+    failed_or_pending_pods = []
+    for pod in pods:
+        if namespace and pod.namespace != namespace:
+            continue
+
+        if pod.exists and pod.instance.status.phase in [
```

Comment:
failed_or_pending_pods = [(pod.name, pod.instance.status.phase) for pod in pods if pod.exists and pod.instance.status.phase in [pod.Status.PENDING,
            pod.Status.FAILED,
        ] ]